math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java

```diff
@@ -14,62 +14,229 @@
 import java.util.stream.Stream;
 
 /**
+ * An undirected graph implementation.
+ * Note that there is no guarantee that the indexes of the vertex or edges are contiguous. To iterate over the vertices
+ * or the edges, it is recommended to use {@link #getVertices()} and {@link #getEdges()}.
+ *
+ * @tparam V The type of the value attached to a vertex.
+ * @tparam E The type of the value attached to an edge.
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public interface UndirectedGraph<V, E> {
 
+    /**
+     * Create a new vertex and notify the {@link UndirectedGraphListener}s.
+     *
+     * @return the index of the new vertex.
+     */
     int addVertex();
 
+    /**
+     * Remove the specified vertex and notify the {@link UndirectedGraphListener}s.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if the vertex doesn't exist or if an edge is connected to this vertex.
+     *
+     * @param v the vertex index to remove.
+     * @return the value attached to the vertex.
+     */
     V removeVertex(int v);
 
+    /**
+     * Return the number of vertices.
+     * @return the number of vertices.
+     */
     int getVertexCount();
 
+    /**
+     * Create an edge between the two specified vertex and notify the {@link UndirectedGraphListener}s.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if one of the vertices doesn't exist.
+     *
+     * @param v1 the first end of the edge.
+     * @param v2 the second end of the edge.
+     * @param obj the value attached to the edge.
+     * @return the index of the new edge.
+     */
     int addEdge(int v1, int v2, E obj);
 
+    /**
+     * Remove the specified edge and notify the {@link UndirectedGraphListener}s.
+     * This method thows a {@link com.powsybl.commons.PowsyblException} if the edge doesn't exist.
+     *
+     * @param e the edge index to remove.
+     * @return the value attached to the edge.
+     */
     E removeEdge(int e);
 
+    /**
+     * Remove all the edges and notify the {@link UndirectedGraphListener}s.
+     */
     void removeAllEdges();
 
+    /**
+     * Return the number of edges.
+     * @return the number of edges.
+     */
     int getEdgeCount();
 
+    /**
+     * Return the indexes of the edges.
```

Comment:
I would say "indices" for the plural form, but it seems indexes is used too. However in technical / math contexts it would mainly be "indices".
https://writingexplained.org/indexes-or-indices-difference